test/integration/replicaset/replicaset_test.go

```diff
@@ -893,6 +895,219 @@ func TestReadyAndAvailableReplicas(t *testing.T) {
 	}
 }
 
+// TestAvailableReplicasWithUnstableReadyCondition tests that replica set correctly updates AvailableReplicas
+// in minReadySeconds time when pods are struggling to become ready.
+// Each test takes approximately 10s to run.
+func TestAvailableReplicasWithUnstableReadyCondition(t *testing.T) {
```

Comment:
Since I am using sleep in this test, I am wondering if we should rather move this either to unit test with a fake the clock or to an e2e test